libs/utils/android/workloads/exoplayer.py

```diff
@@ -0,0 +1,184 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+import logging
+
+from subprocess import Popen, PIPE
+
+from time import sleep
+
+from android import Screen, System, Workload
+
+# Regexps for benchmark synchronization
+
+REGEXPS = {
+    'start'    : '.*Displayed com.google.android.exoplayer2.demo/.PlayerActivity',
+    'duration' : '.*period \[(?P<duration>[0-9]+.*)\]',
+    'end'      : '.*state \[.+, .+, E\]'
+}
+
+class Exoplayer(Workload):
+    """
+    Android Exoplayer workload
+    """
+
+    # Package required by this workload
+    package = 'com.google.android.exoplayer2.demo'
+
+    def __init__(self, test_env):
+        super(Exoplayer, self).__init__(test_env)
+        self._log = logging.getLogger('Exoplayer')
+        self._log.debug('Workload created')
+
+    def _play(self):
+
+        # Grant app all permissions
+        System.grant_permissions(self._target, self.package)
+
+        # Clear logcat
+        self._target.clear_logcat()
+
+        # Handle media file location
+        if not self.from_device:
+            remote_file = self._target.path.join(
+                self._target.working_directory,
+                os.path.basename(self.media_file)
+            )
+
+            self._log.info('Pushing media file to device...')
+            self._target.push(
+                self.media_file,
+                remote_file,
+                timeout = 60
+            )
+            self._log.info('Media file transfer complete')
+        else:
+            remote_file = self.media_file
+
+        # Prepare logcat monitor
+        monitor = self._target.get_logcat_monitor(REGEXPS.values())
+        monitor.start()
+
+        # Play media file
+        play_cmd = 'am start -a "com.google.android.exoplayer.demo.action.VIEW" -d "file://{}"'.format(remote_file)
+        self._log.info(play_cmd)
+        self._target.execute(play_cmd)
+
+        monitor.wait_for(REGEXPS['start'])
+        self.tracingStart()
+        self._log.info('Playing media file')
+
+        line = monitor.wait_for(REGEXPS['duration'])[0]
+        duration = int(round(float(re.search(REGEXPS['duration'], line)
+                                   .group('duration'))))
+
+        self._log.info('Media duration is {}'.format(duration))
+
+        if self.max_duration and self.max_duration < duration:
```

Comment:
I find `max_duration` and `duration` a bit confusing. I'd suggest a simple renaming, something like:

`duration` -> `media_duration_s` (I'd also put the s at the end to say it's in seconds)
`max_duration` -> `play_duration_s` or `play_period_s`